### PR TITLE
Musark 1038

### PR DESCRIPTION
**Pull request by steinaol**
I would appreciate a closer look at the function FieldReadOnly  with some logic to write postfix data, like size and size unit. It seems to cause next instance of FieldReadOnly to be written with a leading space. I cannot figure out why...

### CODE REVIEW

**Review by rituvesh — APPROVED** [review]
LGTM

File: src/modules/sample/SampleEditComponent.jsx [hunk 1]
```diff
@@ -149,20 +150,34 @@ const submitSample = (id:string, appSession: AppSession, form: FormData, editSam
     ...tmpData,
     externalId: {value: tmpData.externalId, source: tmpData.externalIdSource},
     size: {value: tmpData.size, unit: tmpData.sizeUnit},
-    sampleType: {value: tmpData.sampleType, subTypeValue: tmpData.sampleSubType}
+    sampleType: {value: tmpData.sampleType, subTypeValue: tmpData.subTypeValue}
   };
 
+  if (data.size && !data.size.value) {
```
Comment (jarlah) [inline src/modules/sample/SampleEditComponent.jsx]: make helper method in sample model for ex prepareForSubmit or sumtin

File: src/modules/sample/SampleEditComponent.jsx [hunk 2]
```diff
@@ -456,7 +487,11 @@ const SampleEditComponent = ({params, form, updateForm, editSample, appSession}:
           <Button
             onClick={() =>
               submitSample(id, appSession, form, editSample, form.persons.rawValue)
-                .then(() => hashHistory.push(Config.magasin.urls.client.analysis.gotoSample(appSession, id)))
+                .then(() => hashHistory.push(
+                  {
+                  pathname: Config.magasin.urls.client.analysis.gotoSample(appSession, id),
```
Comment (jarlah) [inline src/modules/sample/SampleEditComponent.jsx]: npm run formatcode

File: src/modules/sample/SampleViewComponent.js [hunk 3]
```diff
@@ -15,11 +15,11 @@ import { hashHistory } from 'react-router';
...
 const FieldReadOnly = ({ field, label, postFix }) => (
   <FormGroup>
-    <FormControl.Static>
-      <span className={field.name}>
+    <FormControl.Static className={field.name}>
+      <span>
         <b>{label && `${label}: `}</b>
         {
-          `${field.defaultValue || field.rawValue}${postFix ? ' ' + postFix.defaultValue : ''}`
+          `${field.defaultValue || field.rawValue}${(postFix && ' ' + postFix.defaultValue) || ''}`
```
Comment (jarlah) [inline src/modules/sample/SampleViewComponent.js]: if it doesnt do any good, revert this

**Review by jarlah — CHANGES_REQUESTED** [review]

**Review by jarlah — APPROVED** [review]
Looks good as far as i can see.

File: src/models/sample.js [hunk 4]
```diff
@@ -6,6 +6,36 @@ import { parseISODate, DATE_FORMAT_DISPLAY } from '../shared/util';
 
 class Sample {}
 
+// To clean up after mapping single field to object for backend
+Sample.prepareForSubmit = tmpData => {
```
Comment (ingarabr) [inline src/models/sample.js:10]: Can you rewrite this to a more FP approach. You don't need to do it now but there's some hints in the AnalysisAdd component on how you can solve it.

**Pull request merged**